src/main/java/com/avispl/symphony/dal/communicator/management/zoom/rooms/ZoomRoomsCommunicator.java

```diff
@@ -0,0 +1,581 @@
+     * @throws Exception while ssh communication
+     */
+    private String getMeetingId() throws Exception {
+        String response = execute(ZCOMMAND_CALL_INFO, "*r InfoResult (status=OK)");
+        return parseZoomRoomsProperties(response, "*r InfoResult Info meeting_id").get("*r InfoResult Info meeting_id");
+    }
+
+    /**
+     * Retrieve ZR microphone mute status
+     *
+     * @return true if muted, false if unmuted
+     * @throws Exception during ssh communication
+     */
+    private boolean getMuteStatus() throws Exception {
+        String muteStatus = execute(ZCOMMAND_MUTE_STATUS, "*c zConfiguration Call Microphone Mute").toLowerCase();
+        return muteStatus.substring(muteStatus.indexOf("mute:") + 5, muteStatus.lastIndexOf("** end")).trim().equals(ON);
+    }
+
+    /**
+     * Retrieve ZR camera mute status
+     *
+     * @return true if muted, false if unmuted
+     * @throws Exception during ssh communication
+     */
+    private boolean retrieveCameraMuteStatus() throws Exception {
+        String muteStatus = execute(ZCOMMAND_CAMERA_MUTE_STATUS, "*c zConfiguration Call Camera Mute").toLowerCase();
+        return muteStatus.substring(muteStatus.indexOf("mute:") + 5, muteStatus.lastIndexOf("** end")).trim().equals(ON);
+    }
+
+    /**
+     * Change ZR microphone mute status
+     *
+     * @param status on/off indicating the mute status
+     * @throws Exception during ssh communication
+     */
+    private void switchMuteStatus(boolean status) throws Exception {
+        String meetingStatus = getCallStatus();
+        if (!meetingStatus.equals(IN_MEETING)) {
+            if(logger.isDebugEnabled()){
+                logger.debug("Not in a meeting. Not able to change mute status.");
+            }
+            return;
+        }
+        String command = status ? ON : OFF;
+        executeAndVerify(String.format(ZCOMMAND_MUTE, command), "*c zConfiguration Call Microphone Mute");
+    }
+
+    /**
+     * Change ZR camera mute status
+     *
+     * @param status true or false indicating the mute status
+     * @throws Exception during ssh communication
+     */
+    private void switchCameraMuteStatus(boolean status) throws Exception {
+        String meetingStatus = getCallStatus();
+        if (!meetingStatus.equals(IN_MEETING)) {
+            if(logger.isDebugEnabled()){
+                logger.debug("Not in a meeting. Not able to change camera mute status.");
+            }
+            return;
+        }
+        String command = status ? ON : OFF;
+        executeAndVerify(String.format(ZCOMMAND_CAMERA_MUTE, command), "*c zConfiguration Call Camera Mute");
+    }
+
+    /**
+     * Disconnect from the active call or from an idle call (ZR is trying to connect)
+     *
+     * @return true if disconnected successfully, false if an error has occured
+     */
+    private boolean callDisconnect() throws Exception {
+        meetingNumber = "";

```

Comment:
See my comment to meetingNumber meeting variable.
And in any case, you cannot reset it before you know that your call succeeded.